### PR TITLE
noCache flag should be check before removing csub from cache in mongoUnsubscribeContext()

### PR DESCRIPTION
**Pull request by Anjali-NEC**
Fix issue #2879

### CODE REVIEW

**Comment by fgalan** [comment]
Some test executions has been cancelled due to timeout after passing 6 hours running...

Maybe you have introduced some kind of deadlock in the code that is causing this. I'd suggest to run the tests locally in your environment to see what can be happening.

**Comment by Anjali-NEC** [comment]
> Some test executions has been cancelled due to timeout after passing 6 hours running...
> 
> Maybe you have introduced some kind of deadlock in the code that is causing this. I'd suggest to run the tests locally in your environment to see what can be happening.

Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/c180fa9c51c2e50aea595bd7e1a80f88c0c0f72c if it is ok then please merge it into master

File: src/lib/mongoBackend/mongoUnsubscribeContext.cpp [hunk 1]
```diff
@@ -127,19 +127,21 @@ HttpStatusCode mongoUnsubscribeContext
...
-  if (cSubP != NULL)
-  {
-    subCacheItemRemove(cSubP);
+    if (cSubP != NULL)
+    {
+      subCacheItemRemove(cSubP);
+    }
   }
-
   cacheSemGive(__FUNCTION__, "Removing subscription from cache");
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUnsubscribeContext.cpp:145]: cacheSemGive() should be covered inside the `if (!noCache)` block.

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -1,3 +1,4 @@
+- Fix: noCache flag should be check before removing csub from cache in mongoUnsubscribeContext() (#2879)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
- Hardening: noCache flag should be check before removing csub from cache in mongoUnsubscribeContext() (#2879)
```

Comment (Anjali-NEC) [reply CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/52883254a5a505645d96ab1c31d5b6c00b962b7a

File: src/lib/mongoBackend/mongoUnsubscribeContext.cpp (hunk 1 above)
Comment (Anjali-NEC) [reply src/lib/mongoBackend/mongoUnsubscribeContext.cpp:144]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/52883254a5a505645d96ab1c31d5b6c00b962b7a

File: src/lib/mongoBackend/mongoUnsubscribeContext.cpp [hunk 3]
```diff
@@ -127,20 +127,22 @@ HttpStatusCode mongoUnsubscribeContext
...
-  }
+    CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
 
+    if (cSubP != NULL)
+    {
+      subCacheItemRemove(cSubP);
+    }
   cacheSemGive(__FUNCTION__, "Removing subscription from cache");
+  }
   reqSemGive(__FUNCTION__, "ngsi10 unsubscribe request", reqSemTaken);
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUnsubscribeContext.cpp:146]: Code style:

```suggestion
    cacheSemGive(__FUNCTION__, "Removing subscription from cache");
  }
  
  reqSemGive(__FUNCTION__, "ngsi10 unsubscribe request", reqSemTaken);
```

Comment (Anjali-NEC) [reply src/lib/mongoBackend/mongoUnsubscribeContext.cpp]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/58469edb28b166b8fbf80c01f7e91f7f0fd553b6

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**